src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,667 @@
+                // Find target interface and return dispatcher or null if not found.
+                IntPtr typeMaybe = AsRuntimeDefined(ref riid);
+                if (typeMaybe == IntPtr.Zero)
+                    typeMaybe = AsUserDefined(ref riid);
+
+                return typeMaybe;
+            }
+
+            IntPtr AsRuntimeDefined(ref Guid riid)
+            {
+                for (int i = 0; i < RuntimeDefinedCount; ++i)
+                {
+                    if (RuntimeDefined[i].IID == riid)
+                    {
+                        return Dispatches[i].Vtable;
+                    }
+                }
+
+                return IntPtr.Zero;
+            }
+
+            IntPtr AsUserDefined(ref Guid riid)
+            {
+                for (int i = 0; i < UserDefinedCount; ++i)
+                {
+                    if (UserDefined[i].IID == riid)
+                    {
+                        return Dispatches[i + RuntimeDefinedCount].Vtable;
+                    }
+                }
+
+                return IntPtr.Zero;
+            }
+        }
+        internal unsafe struct EntrySet
+        {
+            public ComInterfaceEntry* Start;
+            public int Count;
+
+            public EntrySet(ComInterfaceEntry* start, int count)
+            {
+                this.Start = start;
+                this.Count = count;
+            }
+        }
+
+        internal unsafe class ManagedObjectWrapperHolder
+        {
+            private ManagedObjectWrapper* wrapper;
+            public ManagedObjectWrapperHolder(ManagedObjectWrapper* wrapper)
+            {
+                this.wrapper = wrapper;
+            }
+
+            public unsafe IntPtr ComIp => this.wrapper->As(ref ComWrappers.IID_IUnknown);
+            ~ManagedObjectWrapperHolder()
+            {
+                Marshal.FreeCoTaskMem((IntPtr)this.wrapper);
+            }
+        }
+
+        internal unsafe struct IUnknownVftbl
+        {
+            public delegate* unmanaged<IntPtr, ref Guid, out IntPtr, int> QueryInterface;
+            public delegate* unmanaged<IntPtr, uint> AddRef;
+            public delegate* unmanaged<IntPtr, uint> Release;
+            public static IUnknownVftbl AbiToProjectionVftbl => ComWrappers.DefaultIUnknownVftbl;
+            public static IntPtr AbiToProjectionVftblPtr => ComWrappers.DefaultIUnknownVftblPtr;
+        }
+
+        /// <summary>
+        /// Globally registered instance of the ComWrappers class for reference tracker support.
+        /// </summary>
+        private static ComWrappers? s_globalInstanceForTrackerSupport;
+
+        /// <summary>
+        /// Globally registered instance of the ComWrappers class for marshalling.
+        /// </summary>
+        private static ComWrappers? s_globalInstanceForMarshalling;
+
+        private static long s_instanceCounter;
+        private readonly long id = Interlocked.Increment(ref s_instanceCounter);
+
+        static unsafe ComWrappers()
```

Comment:
The inline initialization would look better than static constructor. It can be something like:
```
internal static IntPtr DefaultIUnknownVftblPtr { get; } = CreateDefaultIUnknownVftbl();

private static IntPtr CreateDefaultIUnknownVftbl()
{
     IntPtr* vftbl = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(ComWrappers), 3 * sizeof(IntPtr));
     GetIUnknownImplInternal(out vftbl[0], out vftbl[1], out vftbl[2]);
     return (IntPtr)vftbl;
}
```